test.js

```diff
@@ -507,3 +507,20 @@ test('detach child process', async t => {
 
 	t.is(fs.readFileSync(file, 'utf8'), 'foo\n');
 });
+
+// https://github.com/sindresorhus/execa/issues/128
+test('removes exit handler on exit', async t => {
+	// TODO this relies on signal-exit internals
+	const ee = process.__signal_exit_emitter__;
+
+	const child = m('noop');
+	const listener = ee.listeners('exit').pop();
+
+	await new Promise((resolve, reject) => {
+    child.on('error', reject);
+    child.on('exit', resolve);
+	});
+
+	const included = ee.listeners('exit').indexOf(listener) !== -1;
+	t.false(included);
```

Comment:
`t.false(ee.listeners('exit').includes(listener));`

Reply:
`includes` is not supported by Node.js 4.

Putting the expression in `t.false` produces a long and unhelpful message on test failure:

```
/Users/rberdeen/work/execa/test.js:524

   523:
   524:   t.false(ee.listeners('exit').indexOf(listener) !== -1);
   525: });

  Value is not `false`:

  true

  ee.listeners('exit').indexOf(listener) !== -1
  => true

  -1
  => -1

  ee.listeners('exit').indexOf(listener)
  => 15

  listener
  => Function {}

  ee.listeners('exit')
  => [
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
    Function {},
  ]

  ee
  => EventEmitter {
    _events: {
      afterexit: Function {},
      exit: [
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
        Function {},
      ],
    },
    _eventsCount: 2,
    _maxListeners: Infinity,
    count: 2,
    domain: null,
    emitted: {},
    infinite: true,
```

Reply:
Oh, right. `Array.includes` isn't handled by `babel` (in AVA).